### ISSUE TITLE
CPU usage 100% on MacOS

### ISSUE THREAD

**Issue by kim0** [fiddyschmitt/File-Tunnel#24]
`ft` is working very well for me. I'm using it daily. However, I recently noticed the macbook was a little warm, when I checked the `ft` process had one CPU core at 100%. Check the attached screenshot. I'm guessing this might be due to checking if the file was written very rapidly! Maybe there is a way to hook into file monitoring/watch services into macos instead ? or at least check less aggressively. Happy to test any debug builds.
![Screenshot 2024-08-19 at 6 39 57 PM](https://github.com/user-attachments/assets/2acfdac7-8eb0-43f3-8a81-f77fa4859906)

**Comment by fiddyschmitt** [comment]
Thanks Ahmed!
Yes it should definitely not use that much CPU. I'll look into it.

What protocol are you tunneling? Does the high CPU usage happen straight away, after a while, after a disconnect?

**Comment by fiddyschmitt** [comment]
Oh!

I think I found the issue:

```
        public static void Wait(int ms)
        {
            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
            {
                Thread.Sleep(ms);
                return;
            }

            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
            {
                WindowsDelay.Wait(ms);
                return;
            }
        }
```

I'll fix it up tonight :)

**Comment by fiddyschmitt** [comment]
This is now fixed in [v2.2.1](https://github.com/fiddyschmitt/File-Tunnel/releases/tag/v2.2.1)

Thanks Ahmed!
